### DISCUSSION TITLE
How to get register width

### DISCUSSION THREAD

**Discussion by kreuzerkrieg** [jfalcou/eve#1140] [Done & Done!]
From here
https://github.com/jfalcou/eve/issues/1138
why do I need this? since I want to prepare some precomputed data. I can get away with precomputing data for 32 bits, but, since I'm going to operate on, e.g., 256 bits why not to take the advantage?

**Comment by jfalcou** [comment]
How do you precompute them ?
You may want the number of element not bits maybe ?

If you have a type `T`, you can use `eve::expected_cardinal_v<T>` to get the number of element of type T that could fit in a wide.
I

**Reply by kreuzerkrieg** [reply]
For example, when computing CRC32 you dont have (but you can) to compute iteratively crc for each byte since the crc is linear and you can precompute values for each byte followed by zeroes (and then exclusevely OR it), as implemented in Mark Adler's crc32.c. He is working on 32 bit values doing 4 bytes at once (kinda), now, if I have more width, why not to compute 256 bits at once? or 128? or 512? If so, then I have to precompute more tables with ready to go XORed and shifted polynominals. So I have to know the width of my registers to know how many such tables I have to build

**Reply by jfalcou** [reply]
OK so you actually need bits.

You can do

`constexpr auto bits_width = EVE_CURRENT_ABI::bits`;

Sample code: https://godbolt.org/z/eWETv3sEs

**Reply by kreuzerkrieg** [reply]
Looks like exactly what I need! thanks!

**Reply by jfalcou** [reply]
Again, I'm curious to see your final code to see if a better, higher-level process can be used.

**Reply by kreuzerkrieg** [reply]
I bet you do :) Not sure I will achieve what I intend to, it is EVE learning process :)
Meanwhile I will implement it in regular C++ code and then try to convert my final version to EVE, I guess more questions will follow

**Reply by jfalcou** [reply]
Cool, keep us updated :)
